include/aspect/particle/property/interface.h

```diff
@@ -37,6 +37,92 @@ namespace aspect
     namespace Property
     {
 
+      class ParticlePropertyInformation
+      {
+        public:
+          /**
+           * Empty default constructor.
+           */
+          ParticlePropertyInformation();
+
+          /**
+           * Constructor. Initialize the various arrays of this structure with the
+           * given property information collected from the individual plugins.
+           */
+          ParticlePropertyInformation(const std::vector<std::vector<std::pair<std::string,unsigned int> > > &property_information);
+
+          /**
+           * Checks if the particle property specified by @p name exists
+           * in this model.
+           */
+          bool
+          fieldname_exists(const std::string &name) const;
+
+          /**
+           * Get the field index of the particle property specified by @p name.
+           */
+          unsigned int
+          get_field_index_by_fieldname(const std::string &name) const;
+
+          /**
+           * Get the data position of the first component of the particle
+           * property specified by @p name in the property vector of a particle.
+           */
+          unsigned int
+          get_position_by_fieldname(const std::string &name) const;
+
+          /**
+           * Get the number of components of the particle property specified
+           * by @p name.
+           */
+          unsigned int
+          get_components_by_fieldname(const std::string &name) const;
+
+          /**
+           * A vector of all property field names.
+           */
+          std::vector<std::string> field_names;
```

Comment:
I think what this class is missing is a definition of what a component, a property, and a field is. At least I'm a bit confused. I suspect that it is a bit similar to finite element components and blocks, but it would be good to state this concisely somewhere.